packages/lesswrong/components/posts/PostsEditForm.tsx

```diff
@@ -117,8 +126,15 @@ const PostsEditForm = ({ documentId, classes }: {
             if (options?.submitOptions?.redirectToEditor) {
               history.push(postGetEditUrl(post._id, false, post.linkSharingKey));
             } else {
-              history.push({pathname: postGetPageUrl(post)})
-              flash({ messageString: `Post "${post.title}" edited.`, type: 'success'});
+              // If they are publishing a draft, show the share popup
+              // Note: we can't use isDraft here because it gets updated to true when they click "Publish"
+              const showSharePopup = isEAForum && wasEverDraft.current && !post.draft
+              const sharePostQuery = `?${SHARE_POPUP_QUERY_PARAM}=true`
+              history.push({pathname: postGetPageUrl(post), search: showSharePopup ? sharePostQuery : ''})
+
+              if (!showSharePopup) {
+                flash({ messageString: `Post "${post.title}" edited.`, type: 'success'});
```

Comment:
Maybe a nit, but I feel like this (and the other messages) shouldn't have a full stop